### PR TITLE
Replace secp256k1 ECDSA to Ed25519 EdDSA

### PR DESCRIPTION
**Pull request by Trottoir**

### CODE REVIEW

File: cmd/callSC/main.go [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-
-	addr = addr[1:]
-
-	pubKey, err := base58.CheckDecode("zkTGqfwJp43tY4FPgRXC7fr2xML3kDQ8bch15SpnDehuxWiKS")
-	if err != nil {
-		panic(err)
-	}
-
-	privKey, err := base58.CheckDecode("25CHWGN5DZemFnEdPyYfDkyYzEwierr3vCuP3Z4tiChfQpBP41")
+	//TODO use the wallet generated for PB & PK
```
Comment (gregLibert) [inline cmd/callSC/main.go:13]: can you create an issue covering this need and reference here the issue ?

File: pkg/node/sendoperation/sendoperation.go [hunk 2]
```diff
@@ -43,39 +46,28 @@ func message(expiry uint64, fee uint64, senderPubKey []byte, operation Operation
...
 	msg = append(msg, operation.Message()...)
 
 	return msg
 }
 
-func sign(msg []byte, privKey []byte) ([]byte, error) {
-	digest := blake3.Sum256(msg)
-
-	var auxBytes [32]byte
+func sign(msg []byte, pubKey []byte, privKey []byte) ([]byte, error) {
```
Comment (gregLibert) [inline pkg/node/sendoperation/sendoperation.go:55]: It feel weird to pass a public function to a signature : can you append pubKey and msg before calling sign ?

File: pkg/node/sendoperation/sendoperation.go [hunk 3]
```diff
@@ -2,36 +2,39 @@ package sendoperation
 
 import (
 	"context"
-	"crypto/rand"
+	"crypto/ed25519"
 	"encoding/binary"
 	"fmt"
+	"strings"
 
-	"github.com/btcsuite/btcd/btcec/v2"
```
Comment (gregLibert) [inline pkg/node/sendoperation/sendoperation.go:9]: You should perform a `go mod tidy` to remove unused libraries from go.mod and go.sum files.

File: pkg/node/sendoperation/sendoperation.go [hunk 4]
```diff
@@ -2,36 +2,39 @@ package sendoperation
...
 )
 
-type sendOperationReqContent struct {
-	ExpirePeriod int64       `json:"expire_period"`
-	Fee          string      `json:"fee"`
-	Op           interface{} `json:"op"`
-	SenderPK     string      `json:"sender_public_key"`
-}
-
+type JSONableSlice []uint8
```
Comment (gregLibert) [inline pkg/node/sendoperation/sendoperation.go:15]: You shoud put the JSONableSLice just before its MashalJSON method.

File: cmd/callSC/main.go [hunk 5]
```diff
@@ -1,36 +1,24 @@
...
+	"crypto/ed25519"
 	"fmt"
 
 	"github.com/massalabs/thyra/pkg/node"
-	"github.com/massalabs/thyra/pkg/node/base58"
 	sendOperation "github.com/massalabs/thyra/pkg/node/sendoperation"
 	callSC "github.com/massalabs/thyra/pkg/node/sendoperation/callsc"
 )
 
 func main() {
```
Comment (gregLibert) [inline cmd/callSC/main.go:12]: test

File: cmd/callSC/main.go [hunk 6]
```diff
@@ -1,36 +1,24 @@
 package main
```
Comment (Trottoir) [inline cmd/callSC/main.go:1]: This is a test @gregLibert , can you see me ?

Comment (gregLibert) [reply cmd/callSC/main.go:1]: yes

**Review by gregLibert — CHANGES_REQUESTED** [review]
@Trottoir I made some comments on your PR. Can you have a look at them ?

**Review by gregLibert — APPROVED** [review]

**Pull request merged**